File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/roadRunner_1_0/messages/PoseMessage.java
package org.firstinspires.ftc.teamcode.roadRunner_1_0.messages;

import com.acmerobotics.roadrunner.Pose2d;

public final class PoseMessage {
    public static long timestamp;
    public static double x;
    public static double y;
    public static double heading;

    public PoseMessage(Pose2d pose) {
        timestamp = System.nanoTime();
        x = pose.position.x;
        y = pose.position.y;
        heading = pose.heading.toDouble();
    }
}

File: TeamCode/src/main/main/java/org/firstinspires/ftc/teamcode/drivetrain/messages/PoseMessage.kt
package org.firstinspires.ftc.teamcode.drivetrain.messages

import com.acmerobotics.roadrunner.Pose2d

class PoseMessage(pose: Pose2d) {
    var timestamp: Long = System.nanoTime()
    var x: Double = pose.position.x
    var y: Double = pose.position.y
    var heading: Double = pose.heading.toDouble()
}

